### ISSUE TITLE
UndocumentedPublicFunction for functions of an interface inside an internal interface.

### ISSUE THREAD

**Issue by zilinx** [detekt/detekt#3959]
I checked for existing (and closed) issues, but there wasn't anything that matched exactly the problem I want to report. 
I think this only started happening in more recent versions of detekt, but I'm not sure when exactly. I am currently using `1.17.1`

## Expected Behavior of the rule
Detekt should not raise `UndocumentedPublicFunction` for `functions` of an `interface` which is inside an `internal interface`.

## Context
I use this pattern commonly for defining dagger2 components, for example: 
```
@Component(...)
internal interface MyComponent {
    
    @Component.Builder
    interface Builder {
        fun xyz(...): ... 
    }
}
```
In this scenario, `xyz()` is raising `UndocumentedPublicFunction`.
It's not possible to make anything inside an interface explicitly internal, even though it obviously is internal if the outer interface is internal - that's true for both inner interface and its functions. There is no other way than to suppress the rule.
